californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -365,6 +385,15 @@ public UdpMatcher(final NetworkConfig config) {
 			}
 			// ignore response
 			return null;
+		} else if (isResponseRelatedToMulticastRequest(exchange)) {
```

Comment:
> I don't think that we need this separate block

But then only the first response to a multicast request will fall in the block you're suggesting and the next responses will be marked as duplicates, because `prev` variable won't be `null`. Is that what supposed to be?